src/coreComponents/constitutive/fluid/CellBasedFluxSlurryFluidAccessors.hpp

```diff
@@ -0,0 +1,57 @@
+/*	
+ * ------------------------------------------------------------------------------------------------------------	
+ * SPDX-License-Identifier: LGPL-2.1-only	
+ *	
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC	
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University	
+ * Copyright (c) 2018-2020 Total, S.A	
+ * Copyright (c) 2020-     GEOSX Contributors	
+ * All right reserved	
+ *	
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.	
+ * ------------------------------------------------------------------------------------------------------------	
+ */
+
+#ifndef GEOSX_CELLBASEDFLUXSLURRYFLUIDACCESSORS_HPP
+#define GEOSX_CELLBASEDFLUXSLURRYFLUIDACCESSORS_HPP
+
+#include "SlurryFluidExtrinsicData.hpp"
+
+#include "physicsSolvers/fluidFlow/StencilAccessors.hpp"
+
+namespace geosx
+{
+
+namespace ProppantTransportKernels
+{
+
+class CellBasedFluxSlurryFluidAccessorsImpl
+{
+public:
+  CellBasedFluxSlurryFluidAccessorsImpl( ElementRegionManager const & elemManager,
+                                         string const & solverName,
+                                         arrayView1d< string const > const & regionNames,
+                                         arrayView1d< string const > const & materialNames )
+    : m_impl( elemManager, solverName, regionNames, materialNames )
+  { }
+
+  auto density()
```

Comment:
We could definitely argue on returning `auto`.
I would prefer some explicit types here for better understanding.
Also, we could "_doxygen_" the accessors with contextualized information.

Reply:
My preference is to keep the explicit type here, if possible, otherwise the code becomes more difficult to read. If you don't use a smart IDE that tells you what is behind this `auto`, you have to go the file defining the `struct`, read the type, and come back here.

Also, I have not checked that it is the case here, but we should make sure that these accessor names (here,`density`) have the same name as the accessor names defined in the constitutive model for consistency.

Reply:
> If you don't use a smart IDE

Then do! 😉 

The issue here is that the type is like `ElementRegion::ElementViewAccessor< array1d< ... > >`. And you add the `const`, and you add the `view`... But yes, having an explicit type is often better.

> we should make sure that these accessor names (here,`density`) have the same name as the accessor names defined in the constitutive model for consistency.

I'm not sure to fully understand. Here I took the name of the `extrinsicMeshData`.
Is there another check?

Reply:
For the existing accessors, I was just saying that the new getter names you are defining should be consistent with these ones, but that will be easy to check/enforce:
https://github.com/GEOSX/GEOSX/blob/1f36c9006b8d39b4d8785712428b873900b8dc2e/src/coreComponents/constitutive/fluid/MultiFluidBase.hpp#L107
https://github.com/GEOSX/GEOSX/blob/1f36c9006b8d39b4d8785712428b873900b8dc2e/src/coreComponents/constitutive/relativePermeability/RelativePermeabilityBase.hpp#L134
https://github.com/GEOSX/GEOSX/blob/1f36c9006b8d39b4d8785712428b873900b8dc2e/src/coreComponents/constitutive/permeability/PermeabilityBase.hpp#L117

Reply:
Ah OK. 🕵️ 
I didn't check either.

Do you know if a reorganization such as presented would make those accessors disappear? I do not have the big picture on this.

Reply:
Yes, I don't know about using `auto` here. I've used it for convenience while refactoring.
I guess we have multiple options
- Using `auto`
- Using some `decltype`/`result_of`/`etc` like in the proposition
- Reproducing the type like `Accesor< array2d< double > >`... This duplicates the type owned in the `extrinsicMeshData` trait, but it has the benefit of totally hiding this trait pattern (always interesting). If the trait changes, then the compilation would fail so this duplication _cannot go unnoticed_.